src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ConfigSchemaCacheWarmer.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\CacheWarmer;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Config\Definition\ArrayNode;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\Definition\JsonSchemaGenerator;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ContainerBag;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Generate config json schema.
+ */
+final readonly class ConfigSchemaCacheWarmer implements CacheWarmerInterface
+{
+    public function __construct(
+        private KernelInterface $kernel,
+        private ?LoggerInterface $logger = null,
+    ) {
+    }
+
+    public function warmUp(string $cacheDir, ?string $buildDir = null): array
+    {
+        if (!$buildDir || !$this->kernel->isDebug() || !class_exists(JsonSchemaGenerator::class)) {
+            return [];
+        }
+
+        $generator = new JsonSchemaGenerator($buildDir.'/config.schema.json');
+
+        if ($this->kernel instanceof Kernel) {
+            /** @var ContainerBuilder $container */
+            $container = \Closure::bind(function (Kernel $kernel) {
+                $containerBuilder = $kernel->getContainerBuilder();
+                $kernel->prepareContainer($containerBuilder);
+
+                return $containerBuilder;
+            }, null, $this->kernel)($this->kernel);
```

Comment:
The closure seems unnecessarily complicated.
```suggestion
            $container = $kernel->getContainerBuilder();
            $kernel->prepareContainer($container);
```

Reply:
But `getContainerBuilder` is a protected method. This block is copy-paste from [`ConfigBuilderCacheWarmer`](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ConfigBuilderCacheWarmer.php#L51-L69)